tests/st/calicoctl/test_default_pools.py

```diff
@@ -0,0 +1,167 @@
+                                "--cluster-store-opt kv.keyfile=%s " % \
+                                (ETCD_HOSTNAME_SSL, ETCD_CA, ETCD_CERT,
+                                 ETCD_KEY)
+else:
+    ADDITIONAL_DOCKER_OPTIONS = "--cluster-store=etcd://%s:2379 " % \
+                                get_ip()
+
+
+class TestDefaultPools(TestBase):
+
+    @classmethod
+    def setUpClass(cls):
+        # First, create a host to run things in
+        cls.hosts = []
+        cls.host = DockerHost("host",
+                              additional_docker_options=ADDITIONAL_DOCKER_OPTIONS,
+                              start_calico=False)
+        cls.hosts.append(cls.host)
+
+    def setUp(self):
+        try:
+            self.host.execute("docker rm -f calico-node")
+        except CommandExecError:
+            # Presumably calico-node wasn't running
+            pass
+        wipe_etcd(get_ip())
+
+    @parameterized.expand([
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/27",  None),    # Too small
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/32",  None),    # Too small, but legal
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/33",  None),    # Impossible CIDR
+        (False, "CALICO_IPV4POOL_CIDR", "256.0.0.0/24", None),    # Invalid IP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24",  None),    # Typical non-default pool
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/26",  None),    # Smallest legal pool
+        (True,  "CALICO_IPV6POOL_CIDR", "fd00::/122",   None),    # Smallest legal pool
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/123",   None),    # Too small
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/128",   None),    # Too small, but legal
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/129",   None),    # Impossible CIDR
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "cross-subnet"),  # Typical non-default pool, IPIP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "always"),  # Typical non-default pool, IPIP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "off"),  # Typical pool, explicitly no IPIP
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", "always"),  # IPv6 - IPIP not permitted
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", "cross-subnet"),  # IPv6 - IPIP not permitted
+        (True,  "CALICO_IPV4POOL_CIDR", "0.0.0.0/0",    None),    # Valid, but not sensible
+        (True,  "CALICO_IPV6POOL_CIDR", "::/0",         None),    # Valid, but not sensible
+        (True,  "CALICO_IPV6POOL_CIDR", "fd80::0:0/120", None),   # Valid, but non-canonical form
+        (False, "CALICO_IPV6POOL_CIDR", "1.2.3.4/24",    None),   # Wrong type
+        (False, "CALICO_IPV4POOL_CIDR", "fd00::/24",     None),   # Wrong type
+        (True,  "CALICO_IPV6POOL_CIDR", "::0:a:b:c:d:e:0/120", None),  # Valid, non-canonical form
+        (False, "CALICO_IPV4POOL_CIDR", "1.2/16",       None),    # Valid, unusual form
```

Comment:
If that is a valid format this should probably work.  I believe the IP validation we depend on comes from the go library so it might just be a version that it does not support.
Either way I don't think it is a big deal.  We could create an issue around this but I feel like it could be marked 'won't fix' .... or just not worry about it all.